### PR TITLE
LPS-178785 [PRM] Added Partnership Level Fragment to Dashboard

### PR DESCRIPTION
**Pull request by JoMaAlves**

### CODE REVIEW

**Comment by JoMaAlves** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/common/components/level-progress-bar/index.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+			return `${leftToTotal} more project`;
+		}
+
+		return '';
+	};
+
+	const formatValue = (value) => {
+		if (type === ChartTypes.ARR) {
+			return `$${value}K`;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/common/components/level-progress-bar/index.js:35]: same here.

Currency and this K need to be dynamic, to add B or M (Millions and Billions). EVP there is something.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+const PartnerIcon = ({level}) => {
+	if (level === PartnershipLevels.SILVER) {
+		return <SilverPartnerIcon />;
+	}
+	else if (level === PartnershipLevels.GOLD) {
+		return <GoldPartnerIcon />;
+	}
+	else if (level === PartnershipLevels.PLATINUM) {
+		return <PlatinumPartnerIcon />;
+	}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:36]:
```suggestion
	if (level === PartnershipLevels.SILVER) {
		return <SilverPartnerIcon />;
	}

	if (level === PartnershipLevels.GOLD) {
		return <GoldPartnerIcon />;
	}

	if (level === PartnershipLevels.PLATINUM) {
		return <PlatinumPartnerIcon />;
	}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+		<div className="d-flex mb-4">
+			<div
+				className={classNames('d-flex p-0 align-items-center', {
+					'col': !children,
+					'col-3': children,
+				})}
+			>
+				<ClayIconProvider>
+					<CheckIcon />
+				</ClayIconProvider>
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:65]:
```suggestion
				<ClayIconProvider>
					<CheckIcon />
				</ClayIconProvider>
```

It's good to add the ClayIconProvider on main component

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+				{text && <span className="col text-paragraph">{text}</span>}
+			</div>
+
+			{children && <div className="col">{children}</div>}
+		</div>
+	);
+};
+
+const levelProperties = {
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:87]: What the difference between levelProperties and partnerLevelData?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+			}
+		}
+
+		setCompleted(checkedItems);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	const getHeadcount = () => {
+		return `${levelProperties[level].headcount.marketing} Marketing / ${levelProperties[level].headcount.sales} Sales`;
+	};
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:160]:
```suggestion
	const getHeadcount = `${levelProperties[level].headcount.marketing} Marketing / ${levelProperties[level].headcount.sales} Sales`;
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+							level === PartnershipLevels.GOLD,
+						'text-info': level === PartnershipLevels.AUTHORIZED,
+						'text-neutral-7': level === PartnershipLevels.SILVER,
+						'text-neutral-10': level === PartnershipLevels.PLATINUM,
+					})}
+				>
+					{level}
+				</span>
+
+				<span className="font-weight-lighter">&nbsp;Partner</span>
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:179]:
```suggestion
				<span className="font-weight-lighter">Partner</span>
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+			</h3>
+
+			{level !== PartnershipLevels.AUTHORIZED && (
+				<>
+					{level !== PartnershipLevels.SILVER && (
+						<>
+							<CheckBoxItem completed={completed.arr} title="ARR">
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:186]:
```suggestion
			{level !== PartnershipLevels.AUTHORIZED && level !== PartnershipLevels.SILVER && (
				<CheckBoxItem completed={completed.arr} title="ARR">				
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/common/components/level-progress-bar/index.js (hunk 1 above)
Comment (kevenleone) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/common/components/level-progress-bar/index.js:35]: https://github.com/pt-liferay-solutions/liferay-portal/pull/3553#discussion_r1144752310

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/common/components/level-progress-bar/index.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+import {ChartTypes} from '../../enums/chartTypes';
+import {naNToZero} from '../../utils';
+
+const Legend = ({currentValue, total, type}) => {
+	const leftToTotal = Math.floor(total - currentValue);
+
+	const formatLeftToTotal = () => {
+		if (leftToTotal) {
+			if (type === ChartTypes.ARR) {
+				return `$${leftToTotal}k more`;
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/common/components/level-progress-bar/index.js:24]: @JoMaAlves we're using a currency function is MDF dashboard. You can use it.

**Comment by JoMaAlves** [comment]
ci:forward

**Comment by JoMaAlves** [comment]
ci:forward

**Pull request closed without merging**